File: lib/channel.py
#!/usr/bin/env python

import logging

from pixel import *

logger = logging.getLogger("playasign.structure.Channel")


class Channel(object):

    def __init__(self, index, numberOfPixels, type, colorspace):
        self.index      = index
        self.pixels     = []
        self.type       = type
        self.colorspace = colorspace

        for i in range(numberOfPixels):
            self.pixels.append(Pixel(i))

        logger.info("new channel created: " + str(self))

    def __repr__(self):
        return "channel #" + str(self.index) + " (" + str(self.type) + " - " + str(len(self.pixels)) + " pixels - " + str(self.colorspace) +")"

File: PlayaSign/PlayaSign/Channel.swift
//
//  Channel.swift
//  PlayaSign
//
//  Created by Sebastian Thiebaud on 9/5/17.
//

import ObjectMapper

final class Channel: Mappable, CustomStringConvertible {
    var index: Int = .none
    var colorspace: ColorSpace = .rgb
    var pixels: [Pixel] = []

    required init?(map: Map) { }

    func mapping(map: Map) {
        index <- map["index"]
        colorspace <- (map["colorspace"], EnumTransform<ColorSpace>())
        pixels <- map["pixels"]

        pixels.forEach { $0.channel = self }
    }

    var description: String {
        return "Channel #\(index) (\(colorspace)): \(pixels.count) pixels"
    }
}
